src/core/thread/mle.cpp

```diff
@@ -1396,8 +1396,6 @@ otError Mle::AppendCslChannel(Message &aMessage)
     otError       error = OT_ERROR_NONE;
     CslChannelTlv cslChannel;
 
-    VerifyOrExit(Get<Mac::Mac>().GetPanChannel() != Get<Mac::Mac>().GetCslChannel(), OT_NOOP);
```

Comment:
change to only append when csl channel is specified, other than remove the condition here?